server/src/main/java/org/candlepin/controller/CandlepinPoolManager.java

```diff
@@ -2268,7 +2268,8 @@ public void deletePools(Collection<Pool> pools, Collection<String> alreadyDelete
             // Revoke/delete entitlements
             if (!entitlements.isEmpty()) {
                 log.info("Revoking {} entitlements...", entitlements.size());
-                this.entitlementCurator.batchDelete(entitlements);
+                this.entitlementCertificateCurator.deleteByEntitlementIds(entitlementIds);
+                this.entitlementCurator.batchDeleteByIds(entitlementIds);
```

Comment:
The one thing that gives me pause here is that EntitlementCurator is one of a handful of curators that has extra behavior on delete. While I've no issue with doing direct SQL operations in general (though I prefer HQL due to Hibernate being Hibernate), I worry that our auto-commit behavior may cause problems if we don't update the entities still in memory.

Have you done any testing or such to verify that skipping the consistency assignments won't cause problems?

Reply:
Surprisingly, there were no issues when running this in a deployed candlepin with a data dump from prod, but it turns out the unit test failures were because of this omission. I have now added back the extra behavior in addition to using the SQL delete.